gmso/core/topology.py

```diff
@@ -978,6 +979,57 @@ def _reindex_connection_types(self, ref):
         for i, ref_member in enumerate(self._set_refs[ref].keys()):
             self._index_refs[ref][ref_member] = i
 
+    def iter_sites(self, key, value):
+        """Iterate through this topology's sites based on certain attribute and their values.
+
+        Parameters
+        ----------
+        key: str
+            The attribute of the site to look for
+        value:
+            The value that the given attribute should be equal to
+
+        Yields
+        ------
+        gmso.abc.abstract_site.Site
+            The site where getattr(site, key) == value
+        """
+        if key not in Site.__iterable_attributes__:
+
+            raise ValueError(
+                f"`{key}` is not an iterable attribute for Site. "
+                f"To check what the iterable attributes are see gmso.abc.abstract_site module."
+            )
+
+        if value is None:
+            raise ValueError(
+                "Expected `value` to be something other than None. Provided None."
+            )
+
+        for site in self.sites:
+            if getattr(site, key) == value:
+                yield site
+
+    def iter_sites_by_residue_label(self, label):
```

Comment:
Update naming conventions after we figure out site residue specific attribute names.